### ISSUE TITLE
Documentation about training YOLO-9000?

### ISSUE THREAD

**Issue by gan-lin** [pjreddie/darknet#151]
Is there any documentation about training YOLO-9000 (I mean, with WordTree) with my own dataset? What file format should I use?

**Comment by yanchao12122** [comment]
1.Make class name list.
2.Make image set. The filename should contains the object's class. For example,  dog is the class labelof file  123_dog_456.png .
3.Make train/test list. Please use absolute path.
4.Decide which network to use.
5.Download weighs file from this page.
6.Use layers in weights file by  darknet partial CFG_FILE WEIGHTS.weights RESULT.weights HOW_MANY_LAYERS .
7.Edit the DATA_CFG file (xxx.data), replace the train/test file with your own.
8.Start training with  darknet classifier DATA_CFG NET_CFG RESULT.weights 
9.Have a cup of cafe~

**Comment by gan-lin** [comment]
Uh... Sorry, I still can't understand. I want to train YOLO-9000 detector (not the classifier). According to the paper, it should use both classification and detection datasets, and I should describe a tree structure of classes in 9k.tree. How to combine classification and detection data and generate the tree?

**Comment by chancocs** [comment]
I would also like to know how to train the YOLO9000 detector on custom classes and (possibly) tree. If somebody has done it before or knows how to do it, please help. I suspect the explanation given above is for training the classifier.

**Comment by kb1ooo** [comment]
@chancocs I agree, I believe the explanation is for training just a classiification network.

**Comment by adamhrv** [comment]
Same. A guide for training YOLO9000 would be great.

I searched the Google Group and people there are also confused about how to structure data for wordtree, and how to train the hierarchical 9000 framework.

**Comment by yaxiongchi** [comment]
@yanchao12122 ,hi,  I want to train YOLO-9000 detector (not the classifier). According to the paper, it should use both classification and detection datasets, and I should describe a tree structure of classes in 9k.tree. How to combine classification and detection data and generate the tree?

**Comment by kadimakipp** [comment]
https://github.com/AlexeyAB/darknet#using-yolo9000

**Comment by kulkarnikeerti** [comment]
> So, I know this is an old post but it is the first spot on google so I wanted to post a satisfying answer. Follow this [link](https://gitlab.eeecs.qub.ac.uk/40126401/csc4006-EdgeBenchmarking/tree/master/Experiments/YOLO/yolo#using-yolo9000) for detail but 9k.tree is organized:` <label> <parent_it>, if parent_id == -1` then this label doesn't have a parent. The index for the label is its location in the 9k.tree file, and the corresponding 9k.names file. I.E 10 is vain or 'n05302499', in 9k.tree, it is 'n05302499 4', so we can go to the 4th index (5th line), and see in 9k.names it corresponds to the label 'body part' I hope this helps someone out there because yolo9000 and the follow up work are super cool!!!

Hi @RoboNuke , I would like to get my custom data in a tree format and train yolo9000 detector. But I still can't find a proper explanation of how its done(data preparation- tree and training with this tree). And the above link is broken. Can you please guide me to a correct article where its explained properly?

**Comment by sidharthanup** [comment]
Any update on this?
